projects/aca-content/src/lib/components/search/search-save/dialog/edit/saved-search-edit-dialog.component.spec.ts

```diff
@@ -0,0 +1,109 @@
+ */
+
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { of } from 'rxjs';
+import { SavedSearchEditDialogComponent } from './saved-search-edit-dialog.component';
+import { ContentTestingModule, SavedSearch, SavedSearchesService } from '@alfresco/adf-content-services';
+import { provideMockStore } from '@ngrx/store/testing';
+import { Store } from '@ngrx/store';
+import { SnackbarErrorAction, SnackbarInfoAction } from '@alfresco/aca-shared/store';
+import { AppTestingModule } from '../../../../../testing/app-testing.module';
+
+describe('SaveSearchEditDialogComponent', () => {
+  let fixture: ComponentFixture<SavedSearchEditDialogComponent>;
+  let component: SavedSearchEditDialogComponent;
+  let savedSearchesService: SavedSearchesService;
+  let store: Store;
+  let submitButton: HTMLButtonElement;
+  const savedSearchToDelete: SavedSearch = {
+    name: 'test',
+    encodedUrl: '1234',
+    order: 0
+  };
+
+  const dialogRef = {
+    close: jasmine.createSpy('close')
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ContentTestingModule, AppTestingModule],
+      providers: [
+        { provide: MatDialogRef, useValue: dialogRef },
+        provideMockStore(),
+        { provide: SavedSearchesService, useValue: { editSavedSearch: () => of() } },
+        { provide: MAT_DIALOG_DATA, useValue: savedSearchToDelete }
+      ]
+    });
+    dialogRef.close.calls.reset();
+    fixture = TestBed.createComponent(SavedSearchEditDialogComponent);
+    component = fixture.componentInstance;
+    savedSearchesService = TestBed.inject(SavedSearchesService);
+    store = TestBed.inject(Store);
+
+    submitButton = fixture.nativeElement.querySelector('#aca-saved-search-edit-dialog-submit-button');
+  });
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('should not save search if form is invalid', fakeAsync(() => {
+    spyOn(savedSearchesService, 'editSavedSearch').and.callThrough();
+    component.form.controls['name'].setValue('');
+    tick();
+    submitButton.click();
+    expect(savedSearchesService.editSavedSearch).not.toHaveBeenCalled();
+  }));
+
+  it('should save search, show snackbar message and close modal if form is valid', fakeAsync(() => () => {
+    spyOn(savedSearchesService, 'editSavedSearch').and.callThrough();
+    setFormValuesAndSubmit();
+    expect(store.dispatch).toHaveBeenCalledWith(new SnackbarInfoAction('APP.BROWSE.SEARCH.SAVE_SEARCH.EDIT_DIALOG.ERROR_MESSAGE'));
+    expect(dialogRef.close).toHaveBeenCalled();
+  }));
+
+  it('should show snackbar error if there is save error', fakeAsync(() => () => {
+    spyOn(savedSearchesService, 'editSavedSearch').and.throwError('');
+    setFormValuesAndSubmit();
+    expect(store.dispatch).toHaveBeenCalledWith(new SnackbarErrorAction('APP.BROWSE.SEARCH.SAVE_SEARCH.EDIT_DIALOG.SUCCESS_MESSAGE'));
```

Comment:
The test case is for errored save, but the snack bar message being checked is for successful message